### DISCUSSION TITLE
Enter data without using Button In entery box

### DISCUSSION THREAD

**Discussion by inderjeet1980** [TomSchimansky/CustomTkinter#570] [General]
how to insert data in box without using button for exp i have 10 ,20 ,30 this total should come in total box after entering number

**Comment by inderjeet1980** [comment]
from tkinter import *
from PIL import Image


root = Tk()
root.title('TOP Tally')
root.geometry('600x105')
root.iconbitmap(r'C:\Users\ongfx\Desktop\INDERJEET\VS\Logo.ico')




#lable FOR NAME
myLable1 = Label(root,width=7, text='BJP', bg='#ff8400', fg='white',font=('arial',15,'bold'))
myLable2 = Label(root,width=7, text='CONG', bg='blue', fg='white',font=('arial',15,'bold'))
myLable3 = Label(root,width=7, text='JDS', bg='green', fg='white',font=('arial',15,'bold'))
myLable4 = Label(root,width=7, text='OTH', bg='gray', fg='white',font=('arial',15,'bold'))
myLable5 = Label(root,width=7, text='TOTAL', bg='red', fg='white',font=('arial',15,'bold'))

myLable1.grid(padx=15,pady=5, row=0, column=0)
myLable2.grid(padx=15,pady=5, row=0, column=1)
myLable3.grid(padx=15,pady=5, row=0, column=2)
myLable4.grid(padx=15,pady=5, row=0, column=3)
myLable5.grid(padx=15,pady=5, row=0, column=4)


bjp= StringVar()
cong = StringVar()
jds = StringVar()
oth = StringVar()
total = StringVar()


# ENTERY BOX
myLable1Entery = Entry(root,textvariable=bjp,width=7, font=('arial',16,'bold'))
myLable1Entery.grid(row=1,column=0, padx=15)
myLable2Entery = Entry(root,textvariable=cong, width=7, font=('arial',16,'bold'))
myLable2Entery.grid(row=1,column=1, padx=15)
myLable3Entery = Entry(root,textvariable=jds,width=7, font=('arial',16,'bold'))
myLable3Entery.grid(row=1,column=2, padx=15)
myLable4Entery = Entry(root,width=7,textvariable=oth, font=('arial',16,'bold'))
myLable4Entery.grid(row=1,column=3, padx=15)



def add():
    b = int(bjp.get())
    c = int(cong.get())
    d = int(jds.get())
    o = int(oth.get())
    r = b + c + d + o
    total.set(r)



mytotal = Button(root,command=add, text='TOTAL',font=('Gill Sans MT',10,'bold'))
mytotal.grid(row=2, column=4)

myLable5Entery = Entry(root,textvariable=total , width=7,font=('arial',16,'bold'))
myLable5Entery.grid(row=1,column=4, padx=15)



root.mainloop()

**Comment by Akascape** [comment]
@inderjeet1980 
You can use text variables for the CTk entries, then you can trace them whenever their values get changed.
Trace method guide: https://www.geeksforgeeks.org/tracing-tkinter-variables-in-python/

### Here is one example in customtkinter that multiplies two numbers:
```python
import tkinter
import customtkinter

customtkinter.set_appearance_mode("System") 
customtkinter.set_default_color_theme("blue")

HEIGHT = 200
WIDTH = 300
                
app = customtkinter.CTk()
app.title("Dynamic Entry")
app.geometry((f"{WIDTH}x{HEIGHT}"))

label_1 = customtkinter.CTkLabel(master=app,text="Multiply 2 numbers")
label_1.pack(pady=10, padx=10)

def calculate(var, index, mode):
    if entry_1.get()!="" and entry_2.get()!="":
        label_2.configure(text="Total: "+str(int(entry_1.get())*int(entry_2.get())))
    else:
        label_2.configure(text="Total: --")
        
def only_numbers(char):
    #Validate for numbers only
    if ((char.isdigit()) or (char=="")):
        return True
    else:
        return False
    
validation = app.register(only_numbers)

# Make two Interget variables
var1=tkinter.IntVar() 
var2=tkinter.IntVar()

entry_1 = customtkinter.CTkEntry(master=app, textvariable=var1, validate='key', validatecommand=(validation, '%P'))
entry_1.pack(pady=10, padx=10)

entry_2 = customtkinter.CTkEntry(master=app, textvariable=var2, validate='key', validatecommand=(validation, '%P'))
entry_2.pack(pady=10, padx=10)

label_2 = customtkinter.CTkLabel(master=app,text="Total: 0")
label_2.pack(pady=10, padx=10)

# Trace the variables
var1.trace_add('write', calculate)
var2.trace_add('write', calculate)

app.mainloop()          
```

**Reply by inderjeet1980** [reply]
thank you

**Reply by inderjeet1980** [reply]
thank you
